core/primitives/src/ed25519.rs

```diff
@@ -488,17 +479,22 @@ impl Pair {
 	}
 }
 
-impl TypedKey for Public {
-	const KEY_TYPE: KeyTypeId = key_types::ED25519;
+impl CryptoType for Public {
+	const KIND: Kind = Kind::Ed25519;
```

Comment:
would have been nice to get rid of `KIND` altogether.

Reply:
I think we can do this now.